src/clj/rems/api/services/catalogue.clj

```diff
@@ -49,8 +49,8 @@
   (db/set-catalogue-item-enabled! (select-keys command [:id :enabled]))
   {:success true})
 
-(defn set-catalogue-item-archived! [{:keys [id archived]}]
-  (let [{:keys [form resource workflow licenses]} (dependencies-for-catalogue-item id)
+(defn set-catalogue-item-archived! [{:keys [id archived]} user-id]
+  (let [{:keys [form resource workflow licenses]} (dependencies-for-catalogue-item id user-id)
```

Comment:
hmmm an owner might've linked a form owned by `organizationA` to a catalogue item owned by `organizationB`. in that case we shouldn't let an organization owner of `organizationA` archive it (and so on for licenses, workflows, unarchiving, etc.). also, these cases should have tests (to document our decision about them)

Reply:
Okay, so this patch doesn't actually let organization-owner to archive items, user-id is passed here only because get-resource needs it in dependencies.-for-catalogue-item This patch only changes API endpoints /api/resources/ and /api/resources/[id]

Reply:
there should probably be userless versions of get-resource and get-resources that can be used by the archiving logic

Reply:
I actually think linking between organizations should not be allowed.

Reply:
Did not touch this part of the code at this point, so it can be decided later.